test/TestUtilities/Test.Utility/MockServer.cs

```diff
@@ -69,7 +64,37 @@ public MockServer()
         /// </summary>
         public void Start()
         {
-            Listener.Start();
+            // Try creating and starting
+
+            int attempts = 1;
+            do
+            {
+                try
+                {
+                    // tests that cancel downloads and exit will cause the mock server to throw, this should be ignored.
```

Comment:
We should log each attempt to know when the retries are taking place and helping or not. I'm going to be Requesting Changes for this comment.

Reply:
Where do you suggest these are logged?  All of this is in test infrastructure, I'd have to pass around an `ITestOutputHelper` which is a bigger change.  Since they're just tests, I'm not sure if anyone would ever see a log output anyway.

Reply:
We decided to not address this comment for now. We will revisit if we keep seeing issues after the retry logic is merged.